### ISSUE TITLE
Allow to change default path for created xmp files

### ISSUE THREAD

**Issue by MateuszTrN** [darktable-org/darktable#16377]
**Is your feature request related to a problem? Please describe.**
I'm frustrated when I use "preview" app of MacOS and navigate through photos with up/down arrow keys and every other file I got preview of .xmp file like in the video

**Describe the solution you'd like**
it would be nice if darktable would allow to set a path to xmps pattern up - like /${photosFolder}/${xmp_data} - or whatever that wouldn't put xmps in the same folder as the original photos

**Alternatives**
N/A

**Additional context**

https://github.com/darktable-org/darktable/assets/4653046/3767eb65-fc5f-400f-84cd-e05e50cab27f

**Comment by Hpbirkeland** [comment]
Why not just sort by kind in Finder?

<img width="496" alt="Screenshot 2024-02-25 at 13 41 12" src="https://github.com/darktable-org/darktable/assets/58633938/153bda8c-e1dd-4ff7-8c5a-7a49c5499b4d">

**Comment by MateuszTrN** [comment]
> Why not just sort by kind in Finder?
> 
> <img alt="Screenshot 2024-02-25 at 13 41 12" width="496" src="https://private-user-images.githubusercontent.com/58633938/307591468-153bda8c-e1dd-4ff7-8c5a-7a49c5499b4d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4NjYxMTcsIm5iZiI6MTcwODg2NTgxNywicGF0aCI6Ii81ODYzMzkzOC8zMDc1OTE0NjgtMTUzYmRhOGMtZTFkZC00ZmY3LThjNWEtN2E0OWM1NDk5YjRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI1VDEyNTY1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlYzg1MzIzZTBjNzA0ZDJjNDBjZjVmM2UyMWNmZTI0M2ZmZGY4YzJiMTkxMjM5ODc4ZjlmYjAyNDI4ZDJjYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.krcSeOMUVZQZMBllGVwwdy8KItNouhG8uHltDq6R75E">

I do as a workaround, but then other sorting is lost - for example when i have multiple day sessions with everyday review, i like to sort by “date created”

**Comment by darkelectron** [comment]
Does `Finder` have a filter option/method. So you can filter out .xmp.

**Comment by dev-dev-dev-dev-dev-dev-dev-dev** [comment]
What you're describing is a problem with the behavior of that "preview" application - not darktable. Storing XMP sidecar files alongside the image in the same folder is industry-standard behavior.

You can disable XMP files entirely in darktable and instead rely on the darktable database to store image metadata if you so choose. Otherwise consider changes to your workflow (like @darkelectron suggested with filters).

**Comment by danm3d** [comment]
An argument that led to nowhere was held [in another issue](https://github.com/darktable-org/darktable/issues/14022) and I think an important point was brought up, in that new users may find the behaviour of DT and the creation of sidecar files confusing.

I think a simple solution could be to add to the 'never' option's tooltip for creation of sidecar files ([PO for the english string here](https://github.com/darktable-org/darktable/blob/7baaaf1169b0096c816d903d4353cc8233a2b38b/po/en%40truecase.po#L1934)) to explain that changes will still be stored in DT's database, but, by not keeping XMP files on, if the user moves a file to a different directory outside of DT (or causes the file to no longer be found through the db by some other mechanism, the db has no easy way of recovering those changes. 

I've only been using DT for a few weeks, and I was also first puzzled and a bit frustrated when all my raw files suddenly had a weird .xmp file next to it. After reading the docs, I found the setting and am happier to have the option to only create a sidecar file upon edit. But I'm a developer by trade, not a typical user. Someone (a typical user) looking to, for example, ditch the Adobe ecosystem for a FOSS alternative (for ethical or other reasons) might pass on DT because of the frustration/confusion of the XMP files being created in their working dirs with no explanation.

BUT, I also think that it's a lot of effort to build out the feature to store XMPs in a separate directory that won't cause additional frustration, due to user error (see remarks on moving files to different directories above).

So, TL;DR: I think it'd be helpful to display am explainer for the 'never' option in the XMP creation option's tooltip. 

I'm happy to work on the English string and solicit translations (once I find my way around the codebase a bit more) and make a PR for this.

**Comment by ptilopteri** [comment]
> An argument that led to nowhere was held [in another issue](https://github.com/darktable-org/darktable/issues/14022), but I think an important point was brought up, in that new users may find the behaviour of DT and the creation of sidecar files confusing.
> 
> I think a simple solution could be to add to the 'never' option's tooltip for creation of sidecar files ([PO for the english string here](https://github.com/darktable-org/darktable/blob/7baaaf1169b0096c816d903d4353cc8233a2b38b/po/en%40truecase.po#L1934)) to explain that changes will still be stored in DT's database, but, by not keeping XMP files on, if the user moves a file to a different directory outside of DT (or causes the file to no longer be found through the db by some other mechanism, the db has no easy way of recovering those changes.

re-importing the file will attach it's edits from the library.db provided the file name is the same

> I've only been using DT for a few weeks, and I was also first puzzled and a bit frustrated when all my raw files suddenly had a weird .xmp file next to it. After reading the docs, I found the setting and am happier to have the option to only create a sidecar file upon edit. But I'm a developer by trade, not a typical user. Someone (a typical user) looking to, for example, ditch the Adobe ecosystem for a FOSS alternative (for ethical or other reasons) might pass on DT because of the frustration/confusion of the XMP files being created in their working dirs with no explanation.

the fine documentation would have explained that to you.
 
afaik: there is no provision for the xmp files to be stored/accessed automagically in any other location.

**Comment by danm3d** [comment]
> re-importing the file will attach it's edits from the library.db provided the file name is the same

I assumed something like that is in place already. I'm mainly speculating on the "what ifs".

> the fine documentation would have explained that to you.

I sought out the docs, and learned the feature, but not everyone would. I'm talking from a UX standpoint. A change in wording to a tooltip may not seem like much of a difference, but if it meant retaining an extra 0.1% of users who don't read docs, maybe it's worth the trouble?

**Comment by danm3d** [comment]
I think this issue can be closed. @victoryforce has addressed in https://github.com/darktable-org/darktable/pull/16725

**Comment by Pvanuf** [comment]
Gents, sorry to be late at the party, but I was struggling myself with the sidecars. I read here above about to 'rely on the database'. What does that mean? Is non-destructive editing possible without the xmp's?

**Comment by victoryforce** [comment]
> Gents, sorry to be late at the party, but I was struggling myself with the sidecars. I read here above about to 'rely on the database'. What does that mean?

Go to global settings (the gear icon in the bar above the central viewport) -- select the Storage section -- change the value of "create XMP files" from "on import" to "never". While you're doing this, stop to read the tooltip for this setting.


> Is non-destructive editing possible without the xmp's?

All editing in darktable is non-destructive. The XMP (also called sidecar) files are just another place to record image operations, in addition to the library database.

BTW, <ins>destructive</ins> editing of a raw file is nearly impossible in practice. How do you imagine recording edits of a demosaiced image back to a RAW file that contains the original sensor values under the color filter array (i.e. before demosaicing)?

**Comment by Pvanuf** [comment]
Thanks for this. For the time I am using DT just for my iPhone pics. So only jpg’s. So your last remark is indi-hispanic-Chinese to me. But as I said... Thanks

**Comment by macalje** [comment]
Have the same issues but by another reason. I have read everything above and believe I understand that it's not possible. So I just want to show that there is more people with the same need. :)

We use lighttable to rank jpegs in our photo folder. We also wants to watch photos in that folder from other devices, not using DT.

Suggested solution. If the user don't set up DT to store xmp files in a subfolder, DT behaves like it does today. In setup, the user is given the ability so select that DT will store (and look for) xmp files in a subfolder to the folder where the actual picture is.

For me and my family, the solution doesn't need to be more advanced than that.

**Comment by ptilopteri** [comment]
* macalje ***@***.***> [08-20-25 16:12]:
> macalje left a comment (darktable-org/darktable#16377)
> 
> Have the same issues but by another readon. We use lighttable to rank jpegs in our photo folder. We also wants to watch photos in that folder from other devices, not using DT.
> 
> Suggested solution. If the user don't set up DT to store xmp files in a subfolder, DT behaves like it does today. In setup, the user is given the ability so select that DT will store (and look for) xmp files in a subfolder to the folder where the actual picture is.
> 
> For me and my family, the solution doesn't need to be more advanced than that.

why not affect your own solution?  write a script to move the xmp files to
another location when you open a particular session and replace them when your
your session is finished?

or code your own solutiion as none of the dev's appear interested.  it is
FOSS.


-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by macalje** [comment]
Thanks for the suggestion. It is an interesting idea and a good suggestion. Never thought about the possibility. If I had necessary programming skills I would definitely try it out (and some other things🙂). Unfortunate I don't. Just (t)SQL and very basic python, which doesn't help very long in this context... And if it does, this is a good example on how very basic my python knowledge is. But, I do like the program anyway. 😃
